app/src/main/java/com/github/geohunt/app/ui/components/tutorial/Tutorial.kt

```diff
@@ -0,0 +1,240 @@
+import androidx.compose.material3.Button
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.geohunt.app.LoginActivity
+import com.google.accompanist.pager.HorizontalPager
+import com.google.accompanist.pager.rememberPagerState
+import kotlinx.coroutines.launch
+import com.github.geohunt.app.utility.replaceActivity
+import com.google.accompanist.pager.PagerState
+import kotlinx.coroutines.CoroutineScope
+
+/**
+ * The main function for the tutorial that displays the
+ * tutorial slides and the buttons for navigating through
+ *
+ * @param activity The activity that the tutorial is being displayed in
+ */
+@Composable
+fun Tutorial(activity: ComponentActivity) {
+    val items = TutorialSlides.getData()
+    val scope = rememberCoroutineScope()
+    val pageState = rememberPagerState()
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .testTag("Tutorial screen layout")
+    ) {
+        TopButtons(
+            activity = activity,
+            pageState = pageState,
+            scope = scope
+        )
+
+        HorizontalPager(
+            count = items.size,
+            state = pageState,
+            modifier = Modifier
+                .fillMaxHeight(0.9f)
+                .fillMaxWidth()
+                .testTag("Current page is ${pageState.currentPage}"),
+            content = {
+                page -> TutorialSlideContent(items = items[page])
+            },
+        )
+
+        BottomButtons(
+            activity = activity,
+            pageState = pageState,
+            scope = scope,
+            slidesCount = items.size,
+            index = pageState.currentPage,
+            onButtonClick = {
+                if (pageState.currentPage + 1 < items.size) scope.launch {
+                    pageState.scrollToPage(pageState.currentPage + 1)
+                }
+            }
+        )
+    }
+}
+
+/**
+ * Displays the buttons for moving to the next slide
+ * and skipping the tutorial at the top of the screen
+ *
+ * @param activity The activity that the tutorial is being displayed in
+ * @param pageState The state of the pager that the tutorial is using
+ * @param scope The coroutine scope that the tutorial is running in
+ */
+@Composable
+fun TopButtons(
```

Comment:
I think it would make more sense having both the go forward/backward buttons at the bottom of the screen.
You should put the "Go back button" in BottomButtons.

Reply:
I strongly disagree with your suggestion. In general UI design, the go back button is usually placed in the top left side of the interface. Take a watch at any interface, for example in your browser in which you are probably reading this comment now.